internal/math/math.go

```diff
@@ -0,0 +1,11 @@
+package math
+
+import "golang.org/x/exp/constraints"
+
+// Min is a generic min func
+func Min[T constraints.Ordered](a, b T) T {
```

Comment:
This function is used only once and could be easily replaced with 3 lines of code.